provider/fs/src/export/mod.rs

```diff
@@ -31,31 +31,31 @@
 //! let mut exporter = fs_exporter::FilesystemExporter::try_new(serializer, options)
 //!     .expect("Should successfully initialize data output directory");
 //!
-//! // Export a key
-//! let source_provider = HelloWorldProvider::new_with_placeholder_data();
-//! let result = icu_provider::export::export_from_iterable(
+//! // Export something
+//! let payload = DataPayload::<HelloWorldV1Marker>::from_owned(
+//!     HelloWorldV1 { message: Cow::Borrowed("Hi") }
+//! );
+//! let result = exporter.put_payload(
 //!     HelloWorldV1Marker::KEY,
-//!     &source_provider,
-//!     &mut exporter)
+//!     Default::default(),
+//!     payload.clone().into_serializable())
 //! .expect("Should successfully export");
 //!
 //! // Create a filesystem provider reading from the demo directory
-//! let fs_provider = FsDataProvider::try_new(demo_path.clone())
+//! let provider = FsDataProvider::try_new(demo_path.clone())
 //!     .expect("Should successfully read from filesystem");
 //!
 //! // Read the key from the filesystem and ensure it is as expected
 //! let req = DataRequest {
-//!     options: langid!("bn").into(),
+//!     options: Default::default(),
 //!     metadata: Default::default(),
 //! };
-//! let source_response: DataResponse<HelloWorldV1Marker> =
-//!     source_provider.load_resource(&req).unwrap();
-//! let fs_response: DataResponse<HelloWorldV1Marker> =
-//!     fs_provider.load_resource(&req).unwrap();
+//! let response: DataPayload<HelloWorldV1Marker> =
+//!     provider.load_resource(&req).unwrap().take_payload().unwrap();
 //!
 //! assert_eq!(
-//!     source_response.payload.unwrap().get(),
-//!     fs_response.payload.unwrap().get(),
+//!     response.get(),
+//!     payload.get(),
 //! );
 //!
```

Comment:
Issue: why is testdata.postcard changed? Try to make this a no-op change on testdata.postcard.

Reply:
Because there's a key collision and we're now adding the data to the `ZeroMap2d` in a different order :). See #1622